public/components/trace_analytics/components/dashboard/mode_picker.tsx

```diff
@@ -62,6 +62,17 @@ export function DataSourcePicker(props: {
     );
   };
 
+  const updateUrlWithMode = (key: TraceAnalyticsMode) => {
+    const currentUrl = window.location.href.split('?')[0];
+    const queryParams = new URLSearchParams(window.location.search);
```

Comment:
this will only get the standard search params, but not the search param we use with hash router. for example in playground https://playground.opensearch.org/app/observability-traces#/traces?datasourceId=44cc4c50-6f80-11ed-a8ec-958f39714b93

`window.location.search` is `''`, and `currentUrl` is `'https://playground.opensearch.org/app/observability-traces#/traces'`. The `dataSourceId` information becomes lost?